pymc/tests/test_distributions_moments.py

```diff
@@ -1171,28 +1171,10 @@ def normal_sim(rng, mu, sigma, size):
         (np.array([3, 6, 0.5, 0.5]), 2, None, np.array([1, 1, 0, 0])),
         (np.array([30, 60, 5, 5]), 10, None, np.array([4, 6, 0, 0])),
         (
-            np.array([[30, 60, 5, 5], [25, 25, 25, 25]]),
-            10,
+            np.array([[26, 26, 26, 22]]),               # Dim: 1 x 4
+            np.array([[1], [10]]),                      # Dim: 2 x 1
             None,
-            np.array([[4, 6, 0, 0], [4, 2, 2, 2]]),
-        ),
-        (
-            np.array([[25, 25, 25, 25], [26, 26, 26, 22]]),
-            np.array([1, 10]),
-            None,
-            np.array([[1, 0, 0, 0], [2, 3, 3, 2]]),
-        ),
-        (
-            np.array([26, 26, 26, 22]),
-            np.array([1, 10]),
-            None,
-            np.array([[1, 0, 0, 0], [2, 3, 3, 2]]),
-        ),
-        (
-            np.array([[25, 25, 25, 25], [26, 26, 26, 22]]),
-            np.array([1, 10]),
-            2,
-            np.full((2, 2, 4), [[1, 0, 0, 0], [2, 3, 3, 2]]),
+            np.array([[1, 0, 0, 0], [2, 3, 3, 2]]),     # Dim: 2 x 4
```

Comment:
I'm getting an error here: `TypeError: Cannot convert Type TensorType(int64, matrix) (of Variable Elemwise{Cast{int64}}.0) into Type TensorType(int64, (False, True, False))`

Reply:
It seems moment is returning a 3d tensor?

Reply:
You can try to call `get_moment` method directly and evaluate the returned graph, in order to facilitate debugging.

Reply:
When I do what you suggest (in a debug console), I get a 2d array - confusing!
```
get_moment(model["x"]).eval()
array([[1, 0, 0, 0],
       [2, 3, 3, 2]])
```

Reply:
Are you using the latest Aesara version? There was a bug with shapes/casting, that was fixed recently.

Reply:
Okay, so in that example the distribution shape is actually `(2, 1, 4)`, but the returned moment is `(2, 4)`. 
```python
pm.DirichletMultinomial.dist(
    a=np.array([[26, 26, 26, 22]]),  # Dim: 1 x 4
    n=np.array([[1], [10]]),  # Dim: 2 x 1
).eval().shape  # (2, 1, 4)
```

Reply:
I pushed some changes that I think make the mode consistent with the RV shape. Does it look sensible to you?

This was a great edge case you stumbled upon. We could very well have missed it